### DISCUSSION TITLE
Question:  Is there a way to convert the graph of Arpeggio nodes into a state machine?

### DISCUSSION THREAD

**Discussion by JoeLambourne** [textX/Arpeggio#121] [Q&A]
Hi,

textX and Arpeggio look like great tools.   I'm interested in exploiting a tiny subset of the functionality.   Essentially I want to use textX to define some simple grammars which can be represented as state machines.   

I want to find the graph defining the states as nodes and then edges as the valid transitions between these nodes.   Is there any way to do this out of the box?

Thank you for your help.  Great library!

**Comment by igordejanovic** [comment]
Hi @JoeLambourne. There is visualization support for [grammars and trees in Arpeggio](https://textx.github.io/Arpeggio/2.0/debugging/#visualization). Also, grammars and models from textX [can be visualized](https://textx.github.io/textX/3.1/visualization/).

Not sure what you are trying to achieve. If you would like to create a state machine language you can watch [this video tutorial](https://textx.github.io/textX/3.1/tutorials/state_machine/).
It is relatively straightforward to generate dot code from your models.

Moving this issue to GH discussions.
